lib/Form/Processor/ContentFormProcessor.php

```diff
@@ -100,10 +104,23 @@ public function processCancel(FormActionEvent $event)
             return;
         }
 
-        $this->contentService->deleteVersion($data->contentDraft->getVersionInfo());
+        $content = $data->contentDraft;
+        $contentInfo = $content->contentInfo;
+        $versionInfo = $data->contentDraft->getVersionInfo();
+
+        // if there is only one version you have to remove whole content instead of a version itself
+        if (1 === count($this->contentService->loadVersions($contentInfo))) {
+            $location = $this->locationService->loadParentLocationsForDraftContent($versionInfo)[0];
+            $contentId = $location->getContentInfo()->id; // parent location content id
```

Comment:
Hmm, if this is the only point when you needed Location, you actually didn't need it at all. `$contentId` can be obtained from `$contentInfo` which is above this `if`.

Reply:
This is a `$contentId` of parent location this is why I can't use `$contentInfo`. Maybe variable names are a little bit misleading, I will rename these.

Reply:
Ah, right. I need normal :coffee:, not the imitation I drank this morning...